### ISSUE TITLE
Proposal: `run_tests.py` becomes "TAP", delegates to `et test` to find/run tests

### ISSUE THREAD

**Issue by matanlurey** [flutter/flutter#156243]
Today, the `flutter/engine` repository (soon to become the `engine` _folder_ in a merged repo) uses `./testing/run_tests.py` to (a) collect, (b) configure, and (c) run (and report on the results of) test executables across the repository.  It will be non-trivial to replace this with another tool (i.e. `et test`).

My proposal is to move _elements_ from `run_tests.py` to `et`, but keeping `run_tests.py` as the entrypoint (i.e. for CI) until all behavior is either emulated or sufficiently completed in `et` respectively. Even once we do that, we will need a way to create lists of tests that run (i.e. TAP in google3), and this could continue to be `run_tests.py`'s area of ownership until we decide on another format.

In other words, doing something like:

```sh
./testing/run_tests.py --variant host_debug_unopt_arm64 --type engine
```

Is still _useful_, even with a fully functional `et test, because the collection of tests that make up `--type engine` (and potential configuration they need to run, i.e. conditionally wiring up SwiftShader and such) are not tasks that are going to be (directly) supported by `et test`.

We might end up _tweaking_ `run_tests.py` to have arguments similar to `et`, however:

```sh
./testing/run_tests.py --config host_debug_unopt_arm64 --type engine
```

**Comment by jonahwilliams** [comment]
We believe that et is reasonably complete at this point and so I am closing this issue to shrink the backlog. If you disagree, feel free to reopen.
